service/controller/clusterapi/v29/resource/ipam/create.go

```diff
@@ -248,3 +251,38 @@ func getVPCSubnets(ctx context.Context) ([]net.IPNet, error) {
 
 	return results, nil
 }
+
+func (r *Resource) persistAllocatedNetwork(cr cmav1alpha1.Cluster, azs []string) func(ctx context.Context, subnet net.IPNet) error {
+	return func(ctx context.Context, subnet net.IPNet) error {
+		return r.splitAndPersistReservedSubnet(ctx, cr, subnet, azs)
```

Comment:
Would it make sense to just inline that function body here? It's not performing splitting anymore as that will be done by another resource later in the execution.

Reply:
Let's see. I want to talk to you anyway about this thing later.